tests/acceptance/apps_dns/apps_dns_test.go

```diff
@@ -0,0 +1,309 @@
+
+	It("should resolve a k8s service domain name", func() {
+		// Prepare
+		domainName := dns.Fqdn("kubernetes.default.svc.cluster.local")
+		ips := []net.IP{}
+		sdc.Handle(domainName, fake.Handler(ips))
+
+		// Assert
+		By("sending a question type A")
+		res, err := dnsQuery(domainName, dns.TypeA)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(res.Rcode).To(Equal(dns.RcodeSuccess))
+		Expect(res.Answer).ToNot(BeEmpty())
+	})
+
+	Context("resolves an App domain name discovered from the Service Discovery Controller", func() {
+		It("should resolve to a single IPv4", func() {
+			// Prepare
+			domainName := dns.Fqdn(fmt.Sprintf("%d.apps.internal", rand.Uint64()))
+			ips := []net.IP{net.ParseIP("10.11.12.13")}
+			expectedAnswer := []dns.RR{
+				&dns.A{
+					Hdr: dns.RR_Header{
+						Name:     domainName,
+						Rrtype:   dns.TypeA,
+						Class:    dns.ClassINET,
+						Ttl:      ttl,
+						Rdlength: 4,
+					},
+					A: ips[0].To4(),
+				},
+			}
+			sdc.Handle(domainName, fake.Handler(ips))
+
+			// Assert
+			By("sending a question type A")
+			res, err := dnsQuery(domainName, dns.TypeA)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(res.Rcode).To(Equal(dns.RcodeSuccess))
+			Expect(res.Answer).To(Equal(expectedAnswer))
+		})
+
+		It("should resolve to a single IPv6", func() {
+			// Prepare
+			domainName := dns.Fqdn(fmt.Sprintf("%d.apps.internal", rand.Uint64()))
+			ips := []net.IP{net.ParseIP("2001:db8::68")}
+			expectedAnswer := []dns.RR{
+				&dns.AAAA{
+					Hdr: dns.RR_Header{
+						Name:     domainName,
+						Rrtype:   dns.TypeAAAA,
+						Class:    dns.ClassINET,
+						Ttl:      ttl,
+						Rdlength: 16,
+					},
+					AAAA: ips[0],
+				},
+			}
+			sdc.Handle(domainName, fake.Handler(ips))
+
+			// Assert
+			By("sending a question type AAAA")
+			res, err := dnsQuery(domainName, dns.TypeAAAA)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(res.Rcode).To(Equal(dns.RcodeSuccess))
+			Expect(res.Answer).To(Equal(expectedAnswer))
+		})
+
+		It("should resolve to multiple IPv4s", func() {
+			// Prepare
+			domainName := dns.Fqdn(fmt.Sprintf("%d.apps.internal", rand.Uint64()))
+			ips := []net.IP{
+				net.ParseIP("10.11.12.13"),
+				net.ParseIP("10.11.12.14"),
+			}
+			expectedAnswer := []dns.RR{
+				&dns.A{
+					Hdr: dns.RR_Header{
+						Name:     domainName,
+						Rrtype:   dns.TypeA,
+						Class:    dns.ClassINET,
+						Ttl:      ttl,
+						Rdlength: 4,
+					},
+					A: ips[0].To4(),
+				},
+				&dns.A{
+					Hdr: dns.RR_Header{
+						Name:     domainName,
+						Rrtype:   dns.TypeA,
+						Class:    dns.ClassINET,
+						Ttl:      ttl,
+						Rdlength: 4,
+					},
+					A: ips[1].To4(),
+				},
+			}
+			sdc.Handle(domainName, fake.Handler(ips))
+
+			// Assert
+			By("sending a question type A")
+			res, err := dnsQuery(domainName, dns.TypeA)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(res.Rcode).To(Equal(dns.RcodeSuccess))
+			Expect(sortedCopy(res.Answer)).To(Equal(sortedCopy(expectedAnswer)))
```

Comment:
We should be able to use `.To(ConsistOf(…))` rather than do our own sorted copy thing.

Reply:
Thanks. I missed this matcher when I was looking for something like it.

Reply:
Fixed with 12d182b.